### PR TITLE
0.2.0-beta

### PR DESCRIPTION
**Pull request by LimpidCrypto**
## High Level Overview of Change

This is the PR meant to be merged into the `main` branch for the `0.2.0-beta` release. 
The PR also contains some changes for the `develop` branch:
- Removal of incomplete `AsyncWebsocketClient`
- Clean up of the `models/mod.rs`:
  - Move of `TransactionType` to `models/transactions/mod.rs`
  - Move of `RequestMethod` to `models/requests/mod.rs`
  - Move of `AccountSetError` to `models/transactions/account_set.rs`
  - ...
- resolve deprecation warnings
- update crates

---

Following features will be added in `0.2.0-beta` release:
- Request models
- Transaction models
- Ledger models
- Utilize `anyhow` and `thiserror` for models
- Utilities regarding `serde` crate
- Utilities regarding `anyhow` crate

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "xrpl-rust"
-version = "0.1.1"
+version = "0.1.2"
```
Comment (sephynox) [inline Cargo.toml:3]: Version needs to be updated.

Comment (LimpidCrypto) [reply Cargo.toml]: Agreed on `0.2.0-beta`

**Review by sephynox — DISMISSED** [review]
LGTM

**Review by sephynox — APPROVED** [review]

**Comment by LimpidCrypto** [comment]
I've opened a new PR #63 where I squashed merged all these commits so they are signed.

**Comment by LimpidCrypto** [comment]
See #63

**Pull request closed without merging**